prow/lib.sh

```diff
@@ -15,17 +15,26 @@
 # limitations under the License.
 
 function setup_and_export_git_sha() {
-  if [ "${CI:-}" == 'bootstrap' ]; then
-    # Handle prow environment and checkout
-    export USER=Prow
-
-    # Make sure we are in the right directory
-    # Test harness will checkout code to directory $GOPATH/src/github.com/istio/istio
-    # but we depend on being at path $GOPATH/src/istio.io/istio for imports
-    if [[ ! $PWD = ${GOPATH}/src/istio.io/istio ]]; then
-      mv "${GOPATH}/src/github.com/${REPO_OWNER:-istio}" "${GOPATH}/src/istio.io"
-      export ROOT=${GOPATH}/src/istio.io/istio
-      cd "${GOPATH}/src/istio.io/istio" || return
+  if [[ -n "${CI:-}" ]]; then
+    if [[ "${CI:-}" == 'bootstrap' ]]; then
+      # TODO: Remove after update to pod-utils
+
+      # Make sure we are in the right directory
+      # Test harness will checkout code to directory $GOPATH/src/github.com/istio/istio
+      # but we depend on being at path $GOPATH/src/istio.io/istio for imports
+      if [[ ! $PWD = ${GOPATH}/src/istio.io/istio ]]; then
+        mv "${GOPATH}/src/github.com/${REPO_OWNER:-istio}" "${GOPATH}/src/istio.io"
+        export ROOT=${GOPATH}/src/istio.io/istio
+        cd "${GOPATH}/src/istio.io/istio" || return
+      fi
+
+      # Set artifact dir based on checkout
+      export ARTIFACTS_DIR="${GOPATH}/src/istio.io/istio/_artifacts"
+    fi
+
+    if [[ "${CI:-}" == 'prow' ]]; then
+      # Set artifact dir based on checkout
+      export ARTIFACTS_DIR="${ARTIFACTS}"
     fi
 
```

Comment:
Perhaps a final catch all case in case $CI is messed up?

Reply:
We are just setting up ARTIFACTS_DIR here. If CI is messed up then this won't be set.